modin/experimental/engines/omnisci_on_ray/io.py

```diff
@@ -288,3 +338,94 @@ def _prepare_pyarrow_usecols(cls, read_csv_kwargs):
                 raise NotImplementedError("unsupported `usecols` parameter")
 
         return usecols_md
+
+    @classmethod
+    def _read_csv_check_support(
+        cls,
+        read_csv_kwargs: ReadCsvKwargsType,
+    ) -> Tuple[bool, str]:
+        """
+        Check whatever or not passed parameters are supported by current modin.read_csv
+        implementation.
+
+        Parameters
+        ----------
+        read_csv_kwargs: ReadCsvKwargsType
+                Parameters of read_csv function.
+
+        Returns
+        -------
+        bool:
+            Whatever passed parameters are supported or not.
+        str:
+            Error message that should be raised if user explicitely set `engine="arrow"`.
+        """
+        filepath_or_buffer = read_csv_kwargs.get("filepath_or_buffer", None)
+        header = read_csv_kwargs.get("header", "infer")
+        names = read_csv_kwargs.get("names", None)
+        engine = read_csv_kwargs.get("engine", None)
+        skiprows = read_csv_kwargs.get("skiprows", None)
+        delimiter = read_csv_kwargs.get("delimiter", None)
+
+        if isinstance(filepath_or_buffer, str):
+            file_extension = os.path.splitext(filepath_or_buffer)[1]
+            if (
+                not cls.file_exists(filepath_or_buffer, only_locally=True)
+                or file_extension != ".csv"
```

Comment:
Why can't I load a `.txt` file? Does this restriction come from Pandas?

Reply:
`.txt` extension was just missed. Fixed in the last commit.

Reply:
Does Arrow really refuse to parse files with other extensions?

Reply:
Arrow doesn't refuse most of the text formats, but for these formats both Pandas ([Pandas documentation](https://pandas.pydata.org/pandas-docs/stable/user_guide/io.html)) and Arrow ([Arrow documentation](https://arrow.apache.org/docs/python/api/formats.html#json-files)) have specific functions (`read_json` or `read_fwf` for example), so, i think we don't need to handle it here.

Reply:
This is not about formats, this is about file names. Why can't I read CSV data from files named `data.tmp`, `my_csv_file`, etc.?

Reply:
Ok, i see. You can read these files, but in that case (file extension is absent) we lose opportunity to check file type, that can be useful if user passes incorrect file type, for example
```
from pyarrow.csv import read_csv, ReadOptions, ConvertOptions
import pandas


test_filename = "modin/pandas/test/data/test_categories.json"
ans_pandas = pandas.read_csv(test_filename)
print(ans_pandas)
ans_arrow = read_csv(test_filename)

Output:
Empty DataFrame
Columns: [{"one":{"0":111, 1:222, 2:333}, two:{"0":"AAA", 1:"BBB", 2:"CCC"}}]
Index: []
Traceback (most recent call last):
  File "test.py", line 418, in <module>
    ans_arrow = read_csv(test_filename)
  File "pyarrow/_csv.pyx", line 714, in pyarrow._csv.read_csv
  File "pyarrow/error.pxi", line 122, in pyarrow.lib.pyarrow_internal_check_status
  File "pyarrow/error.pxi", line 84, in pyarrow.lib.check_status
pyarrow.lib.ArrowInvalid: CSV parse error: Empty CSV file or block: cannot infer number of columns
```
So it is needed to check file types. To sniff file type we can use third party packages, but i prefer to explicitly prohibit using of files without extensions like it is done in the last commit. What do you think?

Reply:
From here I see that Pandas didn't discover the format and read it as if it were in CSV format. It didn't try to assume JSON format basing on the file extension. There is a `read_json` method to read JSON files.
There is a `compression` param which by default requires us to check for ‘.gz’, ‘.bz2’, ‘.zip’, and ‘.xz’ extensions to infer compression. I'm not sure if Arrow handles it for us. Otherwise, we shouldn't care about the file extension.

Reply:
Ok, extension checks were removed and in order to track such cases (as was showed in the previous comment), `ArrowInvalid` exception was added to the catched exceptions list [here](https://github.com/amyskov/modin/blob/read_csv_omnisci_compatibility/modin/experimental/engines/omnisci_on_ray/io.py#L271).
Arrow infers `compression` automatically according to Arrow [read_csv docstring](https://github.com/apache/arrow/blob/master/python/pyarrow/_csv.pyx#L763-L766).